cooltools/api/dotfinder.py

```diff
@@ -1307,20 +1309,16 @@ def scoring_and_histogramming_step(
     # standard multiprocessing implementation
     if nproc > 1:
         logging.info(f"creating a Pool of {nproc} workers to tackle {len(tiles)} tiles")
-        pool = mp.Pool(nproc)
-        map_ = pool.imap
         map_kwargs = dict(chunksize=int(np.ceil(len(tiles) / nproc)))
     else:
         logging.info("fallback to serial implementation.")
-        map_ = map
         map_kwargs = {}
-    try:
-        # consider using
-        # https://github.com/mirnylab/cooler/blob/9e72ee202b0ac6f9d93fd2444d6f94c524962769/cooler/tools.py#L59
-        histogram_chunks = map_(job, tiles, **map_kwargs)
-    finally:
-        if nproc > 1:
-            pool.close()
+        
+    map_ = map
```

Comment:
this looks confusing to me - perhaps we can rename the name corresponding kwarg in the function itself from `map` to `_map` or `map_` or `local_map` and then just apply it as is without such a renaming `_chunks = local_map(job, tiles, **map_kwargs)`

Reply:
why does it need renaming at all?

Reply:
The `map` parameter should be shadowing the builtin map inside the function. If the builtin version is really needed, all you need to do is `import builtins` and use `builtins.map`.

If this is considered too ugly, then I think the alternative should be an unambiguous name for the function parameter, i.e. what should the `@pool_decorator`-compatible function protocol be: `nproc` + ??? I might suggest something like `map_func` or `map_functor`.

Reply:
+1 for `map_functor` - this make an intent clear, i think - i.e. `map_functor` is a `map` that has been messed with. And get rid of that `map_ = map` inside